extensions/sql-database-projects/src/models/options/deployOptionsModel.ts

```diff
@@ -0,0 +1,962 @@
+
+			case constants.VerifyDeployment:
+				return this.deploymentOptions.verifyDeployment;
+
+			case constants.IgnoreComments:
+				return this.deploymentOptions.ignoreComments;
+
+			case constants.IgnoreColumnCollation:
+				return this.deploymentOptions.ignoreColumnCollation;
+
+			case constants.IgnoreAuthorizer:
+				return this.deploymentOptions.ignoreAuthorizer;
+
+			case constants.IgnoreAnsiNulls:
+				return this.deploymentOptions.ignoreAnsiNulls;
+
+			case constants.GenerateSmartDefaults:
+				return this.deploymentOptions.generateSmartDefaults;
+
+			case constants.DropStatisticsNotInSource:
+				return this.deploymentOptions.dropStatisticsNotInSource;
+
+			case constants.DropRoleMembersNotInSource:
+				return this.deploymentOptions.dropRoleMembersNotInSource;
+
+			case constants.DropPermissionsNotInSource:
+				return this.deploymentOptions.dropPermissionsNotInSource;
+
+			case constants.DropObjectsNotInSource:
+				return this.deploymentOptions.dropObjectsNotInSource;
+
+			case constants.IgnoreColumnOrder:
+				return this.deploymentOptions.ignoreColumnOrder;
+
+			case constants.IgnoreTablePartitionOptions:
+				return this.deploymentOptions.ignoreTablePartitionOptions;
+
+			case constants.DoNotEvaluateSqlCmdVariables:
+				return this.deploymentOptions.doNotEvaluateSqlCmdVariables;
+
+			case constants.DisableParallelismForEnablingIndexes:
+				return this.deploymentOptions.disableParallelismForEnablingIndexes;
+
+			case constants.DisableIndexesForDataPhase:
+				return this.deploymentOptions.disableIndexesForDataPhase;
+
+			case constants.RestoreSequenceCurrentValue:
+				return this.deploymentOptions.restoreSequenceCurrentValue;
+
+			case constants.RebuildIndexesOfflineForDataPhase:
+				return this.deploymentOptions.rebuildIndexesOfflineForDataPhase;
+
+			case constants.PreserveIdentityLastValues:
+				return this.deploymentOptions.preserveIdentityLastValues;
+
+			case constants.IsAlwaysEncryptedParameterizationEnabled:
+				return this.deploymentOptions.isAlwaysEncryptedParameterizationEnabled;
+
+			case constants.AllowExternalLibraryPaths:
+				return this.deploymentOptions.allowExternalLibraryPaths;
+
+			case constants.AllowExternalLanguagePaths:
+				return this.deploymentOptions.allowExternalLanguagePaths;
+
+			case constants.HashObjectNamesInLogs:
+				return this.deploymentOptions.hashObjectNamesInLogs;
+
+			case constants.DoNotDropWorkloadClassifiers:
+				return this.deploymentOptions.doNotDropWorkloadClassifiers;
+
+			case constants.IgnoreWorkloadClassifiers:
+				return this.deploymentOptions.ignoreWorkloadClassifiers;
+
+			case constants.IgnoreDatabaseWorkloadGroups:
+				return this.deploymentOptions.ignoreDatabaseWorkloadGroups;
+
+			case constants.DoNotDropDatabaseWorkloadGroups:
+				return this.deploymentOptions.doNotDropDatabaseWorkloadGroups;
+		}
+		return false;
+	}
+
+	public getDescription(label: string): string {
+		switch (label) {
+			case constants.IgnoreTableOptions:
```

Comment:
Instead of having all these helper functions with huge case statements why don't you create a single mapping table that has the label as the key and an object containing things like the description and functions for updating deployment options? That seems like a much cleaner way of doing it and will make it easier to add more options later if needed.


e.g. Then these methods could just be one-liners like this : 

`return optionsMappingTable[constants.IgnoreTableOptions].description`